### PR TITLE
dts: arm: st: Remove use of CONFIG_SOC_* from STM32 F0 dts files

### PR DESCRIPTION
**Pull request by galak**
To move forward and remove use of Kconfig in dts files lets just create
SoC specific dtsi files that the boards can include.  We also seperate
out the F0 dtsi files into their own dir.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by galak** [comment]
Posted to get feedback on the direction, style, etc.  I've got patches for all the other series, but want to see about the filenames and the use of use 'x' or 'X' in file names.

**Comment by erwango** [comment]
> Posted to get feedback on the direction, style, etc. I've got patches for all the other series, but want to see about the filenames and the use of use 'x' or 'X' in file names.

This is actually coming from you :-).  IIRC, aim was to identify that X stands for unknown, so the dtsi file is common for several variants stm32f030X8 variants (r8, c8).

**Comment by erwango** [comment]
Maybe I proposed this already, but another (cheap) way to do this would be to define DT_FLASH_SIZE and DT_RAM_SIZE directly in board dts files which would avoid the creation of "package" dtsi files.
What do you think?

**Comment by galak** [comment]
> Maybe I proposed this already, but another (cheap) way to do this would be to define DT_FLASH_SIZE and DT_RAM_SIZE directly in board dts files which would avoid the creation of "package" dtsi files.
> What do you think?

I'm not a fan of that since we'd duplicate info if 2 (or more) different boards are using the same specific SoC variant.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_f030r8/nucleo_f030r8.dts [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 
 /dts-v1/;
-#include <st/stm32f030X8.dtsi>
+#include <st/f0/stm32f030X8.dtsi>
```
[the remark this answers is not in the document]
Comment (erwango) [reply boards/arm/nucleo_f030r8/nucleo_f030r8.dts]: Let me rephrase my comment: Let's choose between 'X' or 'x' and apply everywhere. If 'X' is kept this is fine to me also.

**Review by erwango — APPROVED** [review]
